drivers/hmis/app/models/hmis/ce/client_proxy.rb

```diff
@@ -26,8 +26,82 @@ class ClientProxy < GrdaWarehouseBase
 
     scope :for_warehouse_clients, -> { where(client_type: GrdaWarehouse::Hud::Client.sti_name) }
 
+    scope :matching_search_term, ->(search_term) do
+      search_term = search_term.strip
+
+      cp_t = Hmis::Ce::ClientProxy.arel_table
+      c_t = GrdaWarehouse::Hud::Client.arel_table
+      query = cp_t.join(c_t).
+        on(cp_t[:client_id].eq(c_t[:id]).
+        and(cp_t[:client_type].eq('GrdaWarehouse::Hud::Client'))).
+        join_sources
+
+      joins(query).merge(GrdaWarehouse::Hud::Client.text_search(search_term, sorted: false))
+    end
+
+    scope :eligible_for_project_type, ->(project_types) do
+      Rails.logger.info(">>> eligible for #{project_types}")
+      joins(ce_match_candidates: { candidate_pool: { unit_groups: :project, opportunities: :project } }).
```

Comment:
does this need to join to projects through both unit_groups and opportunities? Or could this just go through unit groups?
```suggestion
      joins(ce_match_candidates: { candidate_pool: { unit_groups: :project } }).
```

Reply:
I joined through opportunity too because I _think_ a ClientProxy should be included in "eligible_for_project_type" even if they are only eligible for stale opportunities in the project type. In other words, they belong to a Candidate Pool that is referenced by an Opportunity but not by a Unit Group.